packages/rum-recorder/src/domain/rrweb/observer.ts

```diff
@@ -64,6 +64,7 @@ function initMutationObserver(
     childList: true,
     subtree: true,
   })
+  mutationController.onFreeze(() => mutationBuffer.processMutations(observer.takeRecords()))
```

Comment:
IMO, the relationship between `mutationController`, `mutationBuffer`,  `mutationObserver` and this init function is becoming hard to follow.
FMU:
- mutation controller allow to freeze/unfreeze from other part of the lifecycle
- mutation buffer is not only a buffer but process mutations and emit them
- the onFreeze configuration happen here but the unFreeze configuration happen on the buffer

what about moving the observer to mutationBuffer?
it would allow to encapsulate the observe / process / emit logic, only the freeze would remain deported to the controller.
observer could be disconnected with a stop method on buffer.

With that, I'd be in favor of renaming:
- mutationBuffer to something like mutationObserverWrapper
- mutationController to something like mutationFreezer since it only handle this behavior

Reply:
I moved things like you suggested. I'm mitigated about renaming `MutationController`:

* it is still unclear to me why we need to freeze the mutations during a full snapshot, and I think we could remove parts of this logic in the future

* on top of freeze/unfreeze, upstream now have introduced [lock/unlock](https://github.com/rrweb-io/rrweb/commit/9187bec8148599164cb9e3bc2e6fce15973ecf6e) functions that we should consider adding

Reply:
ok, let's see how it goes